File: program/Medicare/src/main/java/com/Medicare/entity/model/services/ProductService.java
package com.Medicare.entity.model.services;

import java.util.List;
import java.util.Set;

import com.Medicare.entity.model.Product;

public interface ProductService {
public Product createproduct(Product product);
public Product updateproduct(Product product);
public Product getproductbyid(Long productid);
public Set<Product> getallproduct();
public void deleteproductbyid(Long productid);
//get active product
public List<Product> getactiveproduct();
}

File: frontproject/front/src/app/services/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  constructor(private http:HttpClient) { }

  public postproduct(Product:any){
   return this.http.post(`${baseUrl}/product/`,Product);
  }

  public getproduct(){
    return this.http.get(`${baseUrl}/product/`);
  }

  public getproductbyid(productid:any){
    return this.http.get(`${baseUrl}/product/${productid}`)
  }

  public updateeachproduct(productid:any,product:any){
    return this.http.put(`${baseUrl}/product/${productid}`,product)
  }

  public getactiveproduct(){
    return this.http.get(`${baseUrl}/product/active`);
  }

  public deleteproductbyid(productid:any){
    return this.http.delete(`${baseUrl}/product/${productid}`);
  }
}
